packages/react-core/src/components/Popover/PopoverHeader.tsx

```diff
@@ -1,25 +1,48 @@
 import * as React from 'react';
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/react-styles/css/components/Popover/popover';
 import { Title, TitleSizes } from '../Title';
-import { PopoverContext } from './PopoverContext';
+import { PopoverHeaderIcon } from './PopoverHeaderIcon';
+import { PopoverHeaderText } from './PopoverHeaderText';
+
+export interface PopoverHeaderProps extends Omit<React.HTMLProps<HTMLHeadingElement>, 'size' | 'className'> {
+  /** Content of the popover header. */
+  children: React.ReactNode;
+  /** Indicates the header contains an icon. */
+  icon?: React.ReactNode;
+  /** Class to be applied to the header. */
+  className?: string;
+  /** Heading level of the header title */
+  titleHeadingLevel?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+  /** Severity variants for an alert popover. This modifies the color of the header to match the severity. */
+  alertSeverityVariant?: 'default' | 'info' | 'warning' | 'success' | 'danger';
+  /** */
+  id?: string;
+}
 
 export const PopoverHeader: React.FunctionComponent<PopoverHeaderProps> = ({
   children,
+  icon,
+  className,
+  titleHeadingLevel = 'h1',
```

Comment:
I would probably expect this to be `h6` to match the existing `headerComponent` default value.